internal/pkg/composable/providers/kubernetessecrets/kubernetes_secrets.go

```diff
@@ -102,44 +102,61 @@ func getK8sClient(kubeconfig string, opt kubernetes.KubeClientOptions) (k8sclien
 }
 
 // Update the secrets in the cache every RefreshInterval
-func (p *contextProviderK8sSecrets) updateSecrets(ctx context.Context) {
+func (p *contextProviderK8sSecrets) updateSecrets(ctx context.Context, comm corecomp.ContextProviderComm) {
 	timer := time.NewTimer(p.config.RefreshInterval)
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-timer.C:
-			p.updateCache()
+			updatedCache := p.updateCache()
+			if updatedCache {
+				p.logger.Info("Secrets cache was updated, the agent will be notified.")
+				comm.Signal()
```

Comment:
The signal only gets triggered when the cache is updated.

However, the use of the cache is optional. Should we set it as mandatory so we know that in all configurations, we always trigger the signal?

Reply:
Cache is on by default, I think if they disable the cache then its fine that signal is not performed. Without the cache then it cannot perform this action. Would be good to document that behavior, but is acceptable to me.

I didn't see in the unit tests where the call the `comm.Signal` is tested. There is a testing communicator that you can use with `SetOnSignal(func() { ... })`, that will call the function when `comm.Signal` is called. You should test both that signal is called when something is changed, as well as not-called when something is not (using a timeout, is acceptable).

Reply:
I added a unit test to check if the signal gets triggered.